internal/installutil/adoption.go

```diff
@@ -0,0 +1,69 @@
+package installutil
+
+import (
+	"context"
+	"errors"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+	"time"
+)
+
+func AdoptRunner(ctx context.Context, ui terminal.UI, client pb.WaypointClient, id string, addr string) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Waiting for runner to connect to server at %s...", addr)
+	defer func() { s.Abort() }()
+
+	// Waits 5 minutes for the server to detect the new runner before timing out
```

Comment:
How did we decide on 5 min?

Reply:
@xiaolin-ninja this was arbitrarily chosen, but I could see it taking more time depending on the user's environment, which would lead this to unnecessarily time out.

Think we could add a flag like `-adoption-timeout` with a default of 5 mins?

Reply:
I think past discussions have noted that for things that _should_ be snappy, like the server detecting a new runner, 5 minutes is the safe "it really shouldn't take this long at all, if it does, there are problems" length of time.